Fix bugs of "RootElement" mapping results and other issues

XingZhao-CATR opened the pull request:
Based on the last review from @bzeuner , did the following updates:
1. Implement the "presense" mapping;
2. Modify the element naming rule: add a hyphen between characters and
numbers, i.e. "class2" will be mapped into class-2;
3. Sort by the initial letter of the attribute name for elements in "key" list.
For example:

> key 'naming-attribute root-attribute-3';

4. Also, solve the bug that unable to resolve the class/grouping of complex types when used in rpc-parameter. @karthik-sethuraman  Please help review the output :)

XingZhao-CATR committed:
Fix bugs of "RootElement" mapping result and other issues

1. Implement the "presense" mapping;
2. Modify the element naming rule: add a hyphen between characters and
numbers, i.e. "class2" will be mapped into class-2;

XingZhao-CATR committed:
Modify the sorting rule for "key" list

Sort by the initial letter of the attribute name.

bzeuner commented:
Presence Statement is OK.
Element naming rule is fixed.
Elements in "key" list should not be sorted by the initial letter of the attribute name, it should be sorted by the integer value of the isPartOfObjectKey property.
Import Statement is missing in spec-1-model.yang.
Highest Level of Augment path should always be root-instance: "augment "/common-model:root-instance""
Thank you.

XingZhao-CATR commented:
@bzeuner  Thanks for the review.
Will fix above issues asap.

XingZhao-CATR committed:
Merge remote-tracking branch 'OpenNetworkingFoundation/ToolChain' into ToolChain

XingZhao-CATR committed:
Implement "Identity/Base" pattern and solve "RootElement/Specify" related bugs

1. Implement "Identity/Base" pattern for enumeration type;
2. Fix the "key" value sorting rule: sort by the "passByReference"
value, the smaller, the first;
3. Fix the highest Level of Augment path, make it root-instance:
"augment "/common-model:root-instance""
4. Other known issues.
